server/src/service-layer/controllers/PaymentController.ts

```diff
@@ -28,6 +32,48 @@ import {
 
 @Route("payment")
 export class PaymentController extends Controller {
+  @Get("membership_prices")
+  public async getMembershipPrices(): Promise<MembershipStripeProductResponse> {
+    const stripeService = new StripeService()
+    try {
+      const membershipProducts =
+        await stripeService.getActiveMembershipProducts()
+      // Maps the products to the required response type MembershipStripeProductResponse in PaymentResponse
+
+      const productsValues = membershipProducts.map((product) => {
+        // Checks the membership type of the product
+        const membershipType = product.metadata[
+          MEMBERSHIP_TYPE_KEY
+        ] as MembershipTypeValues
+
+        switch (membershipType) {
+          case MembershipTypeValues.UoaStudent:
+          case MembershipTypeValues.NonUoaStudent:
+          case MembershipTypeValues.ReturningMember:
+          case MembershipTypeValues.NewNonStudent:
+        }
+
+        return {
+          productId: product.id,
+          name: product.name,
+          description: product.description,
+          discount: product.metadata.discount === "true",
+          displayPrice: (
+            Number(
+              (product.default_price as Stripe.Price).unit_amount_decimal
+            ) / 100
+          ).toString(),
+          originalPrice: product.metadata.original_price
+        }
+      })
+      return productsValues as MembershipStripeProductResponse
```

Comment:
Dangerous cast, should read the type errors

(Will be something like `{data:productsValues}`)

nit: naming